libcontainer/intelrdt/intelrdt.go

```diff
@@ -370,22 +371,21 @@ func WriteIntelRdtTasks(dir string, pid int) error {
 
 // Check if Intel RDT is enabled
 func IsIntelRdtEnabled() bool {
-	// We have checked the flag before
-	if isIntelRdtEnabled {
-		return true
-	}
+	// We only need to set the flag once
+	once.Do(func() {
```

Comment:
If we only need to do this once, why not just do it in an `init` and set the flag when the program starts?

Reply:
@cyphar 

Thank you for code review.

Intel RDT/CAT is a hardware-dependent feature. We wish add zero/minimal overhead to the hardware platform which doesn't support this feature. So there are several orthogonal call paths in libcontainer to check the capability:
1. In loadFactory() --> libcontainer.New().
2. In LinuxFactory.Create() and Load().
3. In ConfigValidator.Validate()
4. In libcontainer and intelrdt test cases.

Taking loadFactory() --> libcontainer.New() for example. It calls IsIntelRdtEnabled() to check the capability. If not support, libcontainer.New() will not add intelRdtManager as a factory constructor option. In subsequent container creating, running and destroying life cycle, intelRdtManager handles Intel RDT operations. If intelRdtManager is nil, we will not touch Intel RDT related functionality.

From my understanding, keeping IsIntelRdtEnabled() as a "utility" function is no harm in this case. We could check the capability in outside package without adding intelRdtManager hooks. Please correct me if I am wrong.

Thank you.

Reply:
I don't think you understood my question. I was asking why don't we set the `isIntelRdtEnabled` flag inside an `init` rather than setting it when you do the check (and then just making `IsEnabled` check that flag and not touch it). Something like

```
func init() {
  // code to check if it's enabled
  isIntelRdtEnabled = true // or whatever
}

func IsEnabled() {
  return isIntelRdtEnabled
}
```

So you don't need to do the `sync.Do` stuff.

Reply:
@cyphar 
> package intelrdt
> func init() {

I got it. Thank you for suggestion.